src/core/datatypes/phylogenetics/characterdata/AbstractCharacterData.cpp

```diff
@@ -175,11 +175,22 @@ void AbstractCharacterData::addMissingTaxon(const std::string &n) {
  */
 void AbstractCharacterData::addTaxonData(const AbstractTaxonData &obs)
 {
-    
-    // add the sequence name to the list
-    taxa.push_back( obs.getTaxon() );
-    
-    // add the sequence also as a member so that we can access it by name
+	// check if the taxon is already in the data
+	std::map<std::string, AbstractTaxonData* >::iterator it = taxonMap.find( obs.getTaxonName() );
+	if ( it == taxonMap.end() )
+	{
+	    // add the sequence name to the list
+	    taxa.push_back( obs.getTaxon() );
+	}
+	else
+	{
+		// remove the taxon data
```

Comment:
I guess if you add a taxon twice, the data for the first time gets deleted here.
Maybe expand the comment to e.g. `// remove the taxa data for the previous time we added this taxon`
It could also be helpful to print a warning here...

Reply:
I've added a more informative comment.

What's the current Rb approved way of throwing a warning?

Reply:
I think it would be `RBOUT("message\n")`.

(Later I'm going to submit a patch to change this to `RBOUT<<message<<"\n"`)